safe/gui/tools/batch/batch_dialog.py

```diff
@@ -473,12 +476,26 @@ def run_task(self, task_item, status_item, count=0, index=''):
                         reg.instance().addMapLayers(layer_list, False)
                         for layer in layer_list:
                             self.layer_group.addLayer(layer)
+                        # TODO:: Find proper way to recognize layer.
+                        # somehow InaSAFE or QGIS won't recognized added layer
+                        # until we print the layer list in mapCanvas
+                        print "printing layer count and layer list"
+                        print self.iface.mapCanvas().layerCount()
```

Comment:
That's just printing the layer count and the layer list to the python console. 
I tried using `LOGGER.info()`, but it still won't do. The layer is still unrecognised.